rustworkx-core/src/coloring.rs

```diff
@@ -164,44 +368,61 @@ where
 /// use petgraph::graph::NodeIndex;
 /// use petgraph::Undirected;
 /// use rustworkx_core::dictmap::*;
-/// use rustworkx_core::coloring::greedy_node_color;
+/// use rustworkx_core::coloring::{greedy_node_color, GreedyStrategyCore};
 ///
 /// let g = Graph::<(), (), Undirected>::from_edges(&[(0, 1), (0, 2)]);
-/// let colors = greedy_node_color(&g);
+/// let colors = greedy_node_color(&g, GreedyStrategyCore::Degree);
 /// let mut expected_colors = DictMap::new();
 /// expected_colors.insert(NodeIndex::new(0), 0);
 /// expected_colors.insert(NodeIndex::new(1), 1);
 /// expected_colors.insert(NodeIndex::new(2), 1);
 /// assert_eq!(colors, expected_colors);
 /// ```
 ///
-pub fn greedy_node_color<G>(graph: G) -> DictMap<G::NodeId, usize>
+pub fn greedy_node_color<G>(
+    graph: G,
+    greedy_strategy: GreedyStrategyCore,
+) -> DictMap<G::NodeId, usize>
 where
-    G: NodeCount + IntoNodeIdentifiers + IntoEdges,
+    G: NodeCount + IntoNodeIdentifiers + IntoEdges + NodeIndexable,
```

Comment:
Same question here around backwards compatibility? Couldn't this break existing users of `greedy_node_color` because they now have an additional trait constraint on `graph`?

Reply:
I have removed that (apparently this was not needed).